xcube/core/store/ref/store.py

```diff
@@ -0,0 +1,186 @@
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import warnings
+from typing import Iterator, Any, Tuple, Container, Union, Dict, List, \
+    Optional
+
+import fsspec
+import xarray as xr
+
+from xcube.util.jsonschema import JsonObjectSchema
+
+from ..datatype import DataTypeLike
+from ..descriptor import DataDescriptor, DatasetDescriptor
+from ..descriptor import new_data_descriptor
+from ..store import DataStore
+from .schema import REF_STORE_SCHEMA
+
+
+class ReferenceDataStore(DataStore):
+    """A data store that uses kerchunk reference files to
+    open referred NetCDF datasets as if they were Zarr datasets.
+
+    :param refs: The list of reference JSON files to use for this
+        instance. Files (URLs or local paths) are used in
+        conjunction with target_options and target_protocol
+        to open and parse JSON at this location.
+    :param target_protocol: Used for loading the reference files.
+        If not given, protocol will be derived from the given path.
+    :param target_options: Extra filesystem options for loading
+        the reference files.
+    :param remote_protocol: The protocol of the filesystem on which
+        the references will be evaluated. If not given, will be derived
+        from the first URL in the references that has a protocol.
+    :param remote_options: Extra filesystem options for loading the
+        referenced data.
+    :param max_gap: See ``max_block``.
+    :param max_block: For merging multiple concurrent requests to the same
+        remote file. Neighboring byte ranges will only be
+        merged when their inter-range gap is <= `max_gap`.
+        Default is 64KB. Set to 0 to only merge when it
+        requires no extra bytes. Pass a negative number to
+        disable merging, appropriate for local target files.
+        Neighboring byte ranges will only be merged when the
+        size of the aggregated range is <= ``max_block``.
+        Default is 256MB.
+    :param cache_size: Maximum size of LRU cache, where
+        cache_size*record_size denotes the total number of
+        references that can be loaded in memory at once.
+        Only used for lazily loaded references.
+    """
+
+    def __init__(
+            self,
+            refs: List[str],
+            **ref_kwargs,
+    ):
+        self._refs = dict(self._normalize_ref(ref) for ref in refs)
+        self._ref_kwargs = ref_kwargs
+
+    @classmethod
+    def get_data_store_params_schema(cls) -> JsonObjectSchema:
+        return REF_STORE_SCHEMA
+
+    @classmethod
+    def get_data_types(cls) -> Tuple[str, ...]:
+        return ("dataset",)
```

Comment:
```suggestion
        return "dataset",
```

Reply:
I disagree. This is intentional to make clear its a tuple and not a misplaced comma.